File: src/main/java/com/medicalsystem/repository/fieldvalue/FieldValueRepository.java
package com.medicalsystem.repository.fieldvalue;

import com.medicalsystem.model.Patient;
import com.medicalsystem.model.fieldvalue.FieldValue;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface FieldValueRepository<T extends FieldValue<?>> extends JpaRepository<T, Long> {

    List<FieldValue<?>> findAllByPatient(Patient patient);

}

File: src/main/kotlin/com/medicalsystem/repository/value/FieldValueRepository.kt
package com.medicalsystem.repository.value

import com.medicalsystem.domain.Patient
import com.medicalsystem.domain.template.Field
import com.medicalsystem.domain.value.FieldValue
import org.springframework.data.jpa.repository.JpaRepository

interface FieldValueRepository : JpaRepository<FieldValue<*>, Long> {

    fun findAllByPatient(patient: Patient): List<FieldValue<*>>

    fun findByFieldAndPatient(field: Field, patient: Patient): FieldValue<*>?
}